java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -170,6 +170,26 @@ public static <T> Map<String, T[][]> castMapOf2DArray(
                         HashMap::putAll);
     }
 
+    /**
+     * Maps a Map of Object[][] with value type T[][] to value of U[][].
+     *
+     * @param mapOfArrays Map of 2D Array values to cast.
+     * @param clazz The class of the array values to cast to.
+     * @return A Map of arrays of type U[][], containing the key/values from the input Map.
+     * @param <T> The target type which the elements are cast.
+     */
+    public static <T> Map<GlideString, T[][]> castMapOf2DArrayBinary(
```

Comment:
You can update existing function by adding new type parameter
```suggestion
    public static <T> Map<S, T[][]> castMapOf2DArrayBinary(
```
Don't forget to describe it in doc, e.g.
```
@param <S> String type, could be either {@link String} or {@link GlideString}.
```